src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { noop } from 'lodash';
+
+import {
+  Button,
+  Icon,
+  Modal,
+  Pane,
+  PaneMenu,
+  Paneset,
+  SearchField,
+  AccordionSet,
+  Accordion,
+  FilterAccordionHeader,
+  Checkbox,
+} from '@folio/stripes/components';
+import { SearchAndSortSearchButton as FilterPaneToggle } from '@folio/stripes/smart-components';
+
+import { TransformationsForm } from './TransformationsForm';
+import { mappingProfileTransformations } from './TransformationsField/transformations';
+
+import css from './MappingProfilesTransformationsModal.css';
+
+export const MappingProfilesTransformationsModal = ({
+  isOpen,
+  initialTransformationsValues,
+  onCancel,
+}) => {
+  const [isFilterPaneOpen, setFilterPaneOpen] = useState(true);
+
+  const toggleFilterPane = () => {
+    setFilterPaneOpen(curState => !curState);
+  };
+
+  const renderResultsFirstMenu = () => {
+    const hideOrShowMessageId = isFilterPaneOpen
+      ? 'ui-data-export.hideSearchPane'
+      : 'ui-data-export.showSearchPane';
+
+    return (
+      <PaneMenu>
+        <FormattedMessage id={hideOrShowMessageId}>
+          {hideOrShowMessage => (
+            <FilterPaneToggle
+              data-test-transformations-search-toggler
+              aria-label={`${hideOrShowMessage}`}
+              visible={isFilterPaneOpen}
+              onClick={toggleFilterPane}
+            />
+          )}
+        </FormattedMessage>
+      </PaneMenu>
+    );
+  };
+
+  const totalSelectedCount = 0;
+  const searchResultsCount = mappingProfileTransformations.length;
+
+  const footer = (
```

Comment:
Please, could you clarify why you use a constant to create a footer and a function renderResultsFirstMenu to create different parts of the page (inconsistency)? As far as I understand, it is used just to hide hideOrShowMessageId creation. Did I misunderstand the idea?

Reply:
Ah, right. This pattern I borrowed from stripes-acq-components. I will use the function approach as we use such for consistency.